### PR TITLE
fix: use correct secret name when credentials are overridden

### LINKED ISSUE #1666: apps_dns does not find cf-app-sd-client-tls secret credential
**Describe the bug**
This code from @f0rmiga https://github.com/cloudfoundry-incubator/kubecf/blob/master/chart/templates/apps_dns.yaml#L122 specifies that the cf-apps-dns deployment should be looking for a secret with name `var-cf-app-sd-client-tls` . However this feature from @jandubois (https://github.com/cloudfoundry-incubator/kubecf/commit/91e54d7cca320755ddc07a2ac9d5db1b3a21ccc7) means that whenever we specify a value for the cf-app-sd-client-tls ca/certificate/private_key , we end up with the secret being created called `cred-cf-app-sd-client-tls` . We are specifying it with these sections in our values manifest for the helm install

```
credentials:
  cf_app_sd_client_tls.ca: REDACTED
  cf_app_sd_client_tls.certificate: REDACTED
  cf_app_sd_client_tls.private_key
```

**Notice the CRED- prefix instead of the VAR- prefix that the deployment is looking for.** 

Since the VAR- prefixed one doesn't exist, the cf-apps-dns pod won't ever start due to being missing its secret.

Here is the code where @f0rmiga introduced this dependency on the VAR- variable name :

https://github.com/cloudfoundry-incubator/kubecf/commit/329c85bd5a73081404bc6ba2366ced3ceabf5f69#diff-fe38e7f7b88dae1e77e30d919760da6822842ade7353ae51ac5d127881121beaR120

Are we incorrect to specify the secret in this way? I suspect it may not be possible to have this secret specified as a var even,

**To Reproduce**
As explained above

**Expected behavior**
As explained above

**Environment**
cf-operator 7.06
kubecf 2.7.1

**Additional context**

### PR DESCRIPTION
**Pull request by jandubois**
kubecf uses the quarks mechanism to override explicit bosh variables defined in https://github.com/cloudfoundry-incubator/quarks-operator/pull/915

This works automatically for bosh variables, but not for explicit references from manually created kube objects. This PR provides a helper template to generate the proper name.

Fixes #1666

### CODE REVIEW

**Comment by jandubois** [comment]
Just force-pushed this PR because I realized that the template function will work in `mixins/eirini/config/eirini.yaml`. It only requires the user values in `$.Values` and doesn't depend on config files being merged in.

File: chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -84,6 +84,27 @@ app.kubernetes.io/version: {{ default $root.Chart.Version $root.Chart.AppVersion
...
+{{- define "kubecf.varSecretName" }}
+  {{- $root := first . }}
+  {{- $var_name := index . 1 }}
+  {{- $prefix := "var" }}
+  {{- range $name, $value := $root.Values.credentials }}
+    {{- if eq $var_name (splitList "." $name | first) }}
+      {{- $prefix = "cred" }}
+    {{- end }}
+  {{- end }}
+  {{- printf "%s-%s" $prefix $var_name | replace "_" "-" }}
```
Comment (mook-as) [inline chart/templates/_helpers.tpl:105]:
```suggestion
  {{- replace "_" "-" $var_name | printf "%s-%s" $prefix }}
```
… hmm, no, that's not clearer, nevermind.

**Review by mook-as — APPROVED** [review]

File: chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -84,6 +84,27 @@ app.kubernetes.io/version: {{ default $root.Chart.Version $root.Chart.AppVersion
...
+| Returns the name of the kube secret for $var_name. This is normally the variable name
+| prefixed with "var-", unless the value is overridden via $.Values.credentials, in which
+| case it is prefixed with `cred-`.
+==========================================================================================
+*/ -}}
+{{- define "kubecf.varSecretName" }}
+  {{- $root := first . }}
+  {{- $var_name := index . 1 }}
+  {{- $prefix := "var" }}
+  {{- range $name, $value := $root.Values.credentials }}
```
[the remark this answers is not in the document]
Comment (jandubois) [reply chart/templates/_helpers.tpl:100]: @mook-as I see you have already approved the PR and triggered CI. Do you still want me to change this; or should we leave it as-is if CI passes?

[the remark this answers is not in the document]
Comment (mook-as) [reply chart/templates/_helpers.tpl:100]: Leaving it as-is is fine.

**Review by viovanov — APPROVED** [review]

**Pull request merged**